CMakeLists.txt

```diff
@@ -16,13 +16,18 @@ option(
   "Enable substrait-cpp tests. This will enable all other build options automatically."
   ON)
 
-find_package(Protobuf REQUIRED)
-include_directories(${PROTOBUF_INCLUDE_DIRS})
-
-add_subdirectory(third_party)
+# Local files come first.
 include_directories(include)
 include_directories(src)
 
+add_subdirectory(third_party)
+
+find_package(Protobuf QUIET CONFIG)
+if(NOT ${Protobuf_FOUND})
+  find_package(Protobuf REQUIRED)
```

Comment:
Can we add a comment explaining why we need to do this?

Reply:
Done.